src/selectors/profile.js

```diff
@@ -139,6 +139,64 @@ export const getCounterSelectors = (index: CounterIndex): CounterSelectors => {
   return selectors;
 };
 
+const getThreadsForPIDs = createSelector(
+  getThreads,
+  (threads) => {
+    let pid2threads = {};
+    for (let thread of threads) {
+      if (!(thread.pid in pid2threads)) {
+        pid2threads[thread.pid] = [];
+      }
+      pid2threads[thread.pid].push(thread);
+    }
+    return pid2threads;
+  }
+);
+
+export const getIPCCorrelations = createSelector(
+  getThreads,
+  getThreadsForPIDs,
+  (threads, pid2threads) => {
+    function findRecipientMarker(srcData) {
+      if (!(srcData.otherpid in pid2threads)) {
+        throw new Error('recipient not found');
+      }
+      for (let thread of pid2threads[srcData.otherpid]) {
+        for (let i = 0; i < thread.markers.length; i++) {
+          const dstData = thread.markers.data[i];
+          // FIXME: sometimes, `messageseqno`s can be duplicated. We need
+          // stronger guarantees of uniqueness here (or, more likely, in Gecko)!
+          if (dstData && dstData.type === 'IPC' &&
+              dstData.messageseqno === srcData.messageseqno &&
+              dstData.messageid === srcData.messageid &&
+              dstData.direction === 1) {
+            return [thread, i, dstData];
+          }
+        }
+      }
+      throw new Error('recipient not found');
+    }
+
+    let result = {};
```

Comment:
Could this be named `IPCCorrelations`? It's also nice to do an inline type definition here, as Flow will surface errors better if the type is defined at its creation, along with on the function signature.